bin/syclcc-clang

```diff
@@ -530,23 +530,31 @@ class syclcc_config:
           return True
     return False
   
-  def contains_linking_stage(self):
+  def is_preprocessing(self):
     for arg in self.forwarded_compiler_arguments:
       if (arg == "-E" or
           arg == "-fsyntax-only" or
-          arg == "-S" or
-          arg == "-c"):
-        return False
-    return True
+          arg == "-S"):
```

Comment:
I think `-S` also runs the compiler. Would it make more sense to move it to `is_compiling_only`?

Reply:
`-S` should be compiler though not sure what's expected behaviour for single source multitargets.